Tests/AEPOptimizeTests/UnitTests/OptimizePublicAPITests.swift

```diff
@@ -588,8 +588,279 @@ class OptimizePublicAPITests: XCTestCase {
+                XCTAssertEqual(returnedProposition.scope, proposition.scope, "The proposition scope should match the expected value.")
+                
+                /// Verify offers within the proposition
+                XCTAssertEqual(returnedProposition.offers.count, proposition.offers.count, "The number of offers should match.")
+                XCTAssertEqual(returnedProposition.offers.first?.id, proposition.offers.first?.id, "The first offer ID should match.")
+                XCTAssertEqual(returnedProposition.offers.first?.schema, proposition.offers.first?.schema, "The first offer schema should match.")
+            } else {
+                XCTFail("The returned proposition should not be nil.")
+            }
+            
+            expectation.fulfill()
+        }
+        /// wait
+        wait(for: [expectation], timeout: 5)
+    }
+    
+    func testGetProposition_responseExceedsTimeout() {
+        /// setup
+        let decisionScope = DecisionScope(name: "eyJhY3Rpdml0eUlkIjoieGNvcmU6b2ZmZXItYWN0aXZpdHk6MTExMTExMTExMTExMTExMSIsInBsYWNlbWVudElkIjoieGNvcmU6b2ZmZXItcGxhY2VtZW50OjExMTExMTExMTExMTExMTEifQ==")
+        
+        let expectation = XCTestExpectation(description: "getPropositions should timeout as the response is delayed.")
+        expectation.assertForOverFulfill = true
+        
+        let testEventData: [String: Any] = [
+            "requesttype": "getpropositions",
+            "decisionscopes": [
+                [ "name": decisionScope.name ]
+            ]
+        ]
+        
+        let testEvent = Event(name: "Optimize Get Propositions Request",
+                              type: EventType.optimize,
+                              source: EventSource.requestContent,
+                              data: testEventData)
+        
+        /// Mock listener with delayed response
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(
+            type: testEvent.type,
+            source: testEvent.source) { event in
+                DispatchQueue.global().asyncAfter(deadline: .now() + 4) {
+                    /// Delay response beyond timeout
+                    let eventData: [String: Any] = [:]
+                    let responseEvent = event.createResponseEvent(name: OptimizeConstants.EventNames.OPTIMIZE_NOTIFICATION,
+                                                                  type: EventType.optimize,
+                                                                  source: EventSource.notification,
+                                                                  data: eventData)
+                    MobileCore.dispatch(event: responseEvent)
+                }
+            }
+        
+        /// execute
+        Optimize.getPropositions(for: [decisionScope], timeout: 2) { data, error in
+            /// Verify
+            XCTAssertNil(data)
+            XCTAssertNotNil(error)
+            if let error = error as? AEPError {
```

Comment:
this should also be a guard - if `error` is not an `AEPError`, this test will still pass